api/builtins/builtins.go

```diff
@@ -0,0 +1,75 @@
+package builtins
+
+import (
+	internal "sigs.k8s.io/kustomize/api/internal/builtins"
+)
+
+// DEPRECATED: Please stop importing the functions and types below. They will be deprecated in Kustomization v1.
```

Comment:
From [godocs](https://go.dev/blog/godoc): 

> To signal that an identifier should not be used, add a paragraph to its doc comment that begins with “Deprecated:” followed by some information about the deprecation.

For example:

```
// Deprecated: Package api/builtins will not be available in api v1. 
package builtins
```

I think the comment needs to be directly above `package builtins`, to be part of the package's doc comment. This will help linters and IDEs flag uses of these vars/types as deprecated, and will also show up in the [docs here](https://pkg.go.dev/sigs.k8s.io/kustomize/api@v0.10.1/builtins).